it/java/basic/src/test/java/BasicAPITest.java

```diff
@@ -22,11 +22,10 @@ void basicTest() throws Exception {
         );
 
         Assertions.assertNotNull(exception.getCause());
-        Assertions.assertTrue(exception.getCause() instanceof apisdk.models.Error);
-        var error = (apisdk.models.Error) exception.getCause();
+        Assertions.assertTrue(exception.getCause() instanceof com.microsoft.kiota.ApiException);
```

Comment:
@baywet here I changed the code to get the tests passing, but this looks like a regression.
wdyt?

Reply:
I think this has to do with the changes you have implemented in kiota java a little while ago and how the exceptions are built. I could be wrong though.

Reply:
I do believe that this is different, I changed how the `ApiException` is constructed, in this case, seems like the `Error` is not correctly mapped to the definition instead.

Reply:
would you mind looking into this further? I think you're right actually, we're not seeing that issue on other pull requests, so it's probably related to the changes this PR brings

Reply:
I'm convinced that it's the bump of `kiota-java` libs(0.6 -> 0.7), but, ok I can try to have a look probably tomorrow.

Reply:
I think I identified the bug, and you were right it was in my PR 😊 , fixing it 👍

Reply:
fixed here: https://github.com/microsoft/kiota-java/pull/742